include/mgmt/fmfu_mgmt.h

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+/** @file fmfu_mgmt.h
+ * @defgroup fmfu_mgmt MCUMgr module for full Modem Firmware Upgrade
+ * @{
+ * @brief MCUMgr based Full Modem Firmware Update(FMFU).
+ *
+ * The Full Modem Firmware Upgrade (FMFU) is a module with functions
+ * that hooks into MCUMgr for doing a full modem update over a transport layer
+ * with the SMP protocol.
+ *
+ * The command values are 0 for get_hash and 1 for firmware_upload
+ *
+ */
+
+#ifndef MGMT_FMFU_H__
+#define MGMT_FMFU_H__
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/* Define full modem update MCUMgr commands */
+#define FMFU_MGMT_ID_GET_HASH 0
+#define FMFU_MGMT_ID_UPLOAD 1
+
+
+/*
+ * Calculate the MTU through buffer size of mcumgr and subtracting overhead
+ * of the RX buffer.
+ */
+#define SMP_PACKET_MTU\
+	(CONFIG_MCUMGR_BUF_SIZE - \
+	((CONFIG_UART_MCUMGR_RX_BUF_SIZE/CONFIG_MCUMGR_BUF_SIZE) + 1) * 4)
+/*
+ * Define the buffer sizes used for full modem update SMP server
+ * SMP uart buffer is the same as the buffer configured as the RX buffer for
+ * MCUMgr
+ */
+#define SMP_UART_BUFFER_SIZE CONFIG_UART_MCUMGR_RX_BUF_SIZE
+
+
+/** @brief Setup and register the full modem update SMP based full modem
+ *  update command handler group.
```

Comment:
```suggestion
/** @brief Setup and register the command handler group for full modem firmware
 *              update over SMP.
```

Beware of incorrect indentation.